include/envoy/common/optref.h

```diff
@@ -34,6 +34,44 @@ template <class T> struct OptRef : public absl::optional<std::reference_wrapper<
     const T& ref = **this;
     return &ref;
   }
+
+  /**
+   * Helper to convert a OptRef into a pointer. If the optional is not set, returns a nullptr.
+   */
+  T* ptr() {
+    if (this->has_value()) {
+      T& ref = **this;
+      return &ref;
+    }
+
+    return nullptr;
+  }
+
+  /**
+   * Helper to convert a OptRef into a pointer. If the optional is not set, returns a nullptr.
+   */
+  const T* ptr() const {
+    if (this->has_value()) {
+      const T& ref = **this;
+      return &ref;
+    }
+
+    return nullptr;
+  }
+
+  T& ref() { return **this; }
+
+  const T& ref() const { return **this; }
 };
 
+template <class T> OptRef<T> makeOptRef(T& ref) { return {ref}; }
+
+template <class T> OptRef<T> makeOptRef(T* ptr) {
```

Comment:
this overload feels like it might result in ambiguity at some point in the future (thought it would have to be some kind of pathological case on optional reference to a pointer to get into that state, I think). Maybe call this one `makeOptRefFromPtr`?

Reply:
Yeah good point, I'll make the name change